add SEOR code-less smart contract platform application

AKACoder opened the pull request:
# Grant Application Checklist

- [x] The [application-template.md](https://github.com/w3f/Open-Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( "project_name.md") and updated.
- [x] A BTC or Ethereum address for the payment of the milestones is provided inside the application.  
- [x] The software of the project will be released under the Apache license version 2.0 as specified in the [terms and conditions](https://gist.github.com/Noc2/75bc58e8ce9b5d419ff883b0cf2b8c19).
- [x] The total funding amount of the project is below USD $30k for initial grant applications and $100k for follow-up grants at the time of submission.
- [x] The initial PR contains only one commit (squash if needed before submitting your PR).
- [x] The grant will only be announced once we successfully [delivered](https://github.com/w3f/Grant-Milestone-Delivery) the first milestone.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/w3f/Open-Grants-Program?pullRequest=205) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/w3f/Open-Grants-Program?pullRequest=205) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/w3f/Open-Grants-Program?pullRequest=205) it.</sub>

Noc2 commented on applications/SEOR-code-less-smart-contract-platform.md:
```diff
@@ -0,0 +1,145 @@
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 1. | SDK code repository | A git repository of SEOR-JS-SDK. The SDK uses a unified interface to support the interaction between front-end DApp and substrate nodes, Ethereum nodes and other blockchain nodes of different architectures, allowing developers to access different chains with minimal changes without modifying the logic.  |
```
Could you add additional information here (e.g. programming language)?

Noc2 replied:
Also unit-tests would be nice

Noc2 commented on applications/SEOR-code-less-smart-contract-platform.md:
```diff
@@ -0,0 +1,145 @@
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 1. | Platform DApp | A code-less smart contract platform DApp will be deployed under a second-level domain name of this project. Through this DApp, users can deploy the contract implemented in Milestone 1 without coding, to the parachains supporting smart contracts or directly to Ethereum, and contract interaction can be carried out in the platform. |
```
Please also add more technical details here.

Noc2 reviewed (changes requested):
Thanks for the open grants application. I added a few comments below.

AKACoder commented:
@Noc2 Thanks for the review and feedback!

The deliverables have been updated.  

We added licenses and unit test items for each milestone, and made more detailed and technical of the specification.   

Please review.

Noc2 reviewed (approved):
Thanks for the update. Looks good to me and I will share it with the rest of the team.

mmagician commented on applications/SEOR-code-less-smart-contract-platform.md:
```diff
@@ -0,0 +1,155 @@
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 1. | License | Apache License 2.0 |
+| 2. | Contract set Repository | A git repository containing the source code of 4 contracts: parameterized ERC20, liquidity mining, contract deployer and Uniswap protocol connector. These smart contracts are all developed to realizes parameterized during deployment, not copy the standard contract such as ERC20 directly. All of contracts will be will be implemented using both solidity and ink! tec ...
```
Could you provide examples of which variables in the [ERC20](https://github.com/paritytech/ink/blob/v3.0.0-rc2/examples/erc20/lib.rs) contract you would parametrise? At least for this specific contract, there doesn't seem to be much wiggle room for specifying parameters.

AKACoder replied:
Our main idea is to be able to parametrically transform the general business process of smart contract and provide the deployment without coding, instead of modifying the contract code every time when some general business to be implemented.

Taking ERC20 as an example, in addition to the configurable items that have been defined in the standard, we also support: mintable switch (whether mint is allowed),  supply formula and formula parameters setting (used to determine the supply curve of mintable token, such as linear, staged, etc.),  roles and permissions setting (Gov, Admin, Minter, etc.), and blacklist parameter configuration.  

Those parametric design of ERC20 was not imagined out of air. They are common customization requirements of ERC20 standards of many projects we have worked on.

mmagician commented on applications/SEOR-code-less-smart-contract-platform.md:
```diff
@@ -0,0 +1,155 @@
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 1. | License | Apache License 2.0 |
+| 2. | Contract set Repository | A git repository containing the source code of 4 contracts: parameterized ERC20, liquidity mining, contract deployer and Uniswap protocol connector. These smart contracts are all developed to realizes parameterized during deployment, not copy the standard contract such as ERC20 directly. All of contracts will be will be implemented using both solidity and ink! tec ...
```
Furthermore, are you referring to a github repository by `Contract set Repository`? So that means in the future, the community could add additional parametrised contracts, right?

AKACoder replied:
Yes, our current plan is to create a github repository under this organization: https://github.com/SealSC, but due to our initial code is not yet ready, it has not yet been actually created.  

We will accept the new contract code under this repository and handle issues, PRs and proposals from the community and any other interested third parties.

mmagician commented on applications/SEOR-code-less-smart-contract-platform.md:
```diff
@@ -0,0 +1,155 @@
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 1. | License | Apache License 2.0 |
+| 2. | Code-less smart contractPlatform DApp | A code-less smart contract platform DApp will be deployed under a second-level domain name of this project. The DApp's UI part will be implement by VUE, and it will have a custom designed UI to uniform different chain's interface. The interactive with these chains will be supported by SEOR-JS-SDK. Through this DApp, users can deploy the contract impl ...
```
Deploying smart contracts to public chains involves paying a Tx fee. Unless you're going to sponsor such deployments, users will need a way to connect to a wallet (multiple wallets if deploying to multiple chains?) - how do you envision this integration?

AKACoder replied:
We have no plans to pay Tx fees for users, so users must using different wallet  to pay the fee for different chains. 
We will support multiple wallets through a set of unified interface SDK wich implemented in Mile stone-2 and handle the interaction with the chain.  

In fact, we have previously developed a set of unified interface SDK, which supports wallet protocols of different public chains such as ETH, EOS, and ONT. However, due to various reasons, this SDK has not been maintained continuously.

This is the previously SDK documents:
[protocol](https://sealsc.github.io/documents/en/web-extensions/protocol/)
[proxy](https://sealsc.github.io/documents/en/web-extensions/proxy/)
[metamask-extension](https://sealsc.github.io/documents/en/web-extensions/metamask/)
[scatter-extension](https://sealsc.github.io/documents/en/web-extensions/scatter/)
[cyano-extension](https://sealsc.github.io/documents/en/web-extensions/cyano/)

Based on this SDK, we will use TypeScript to rewrite and improve this SDK in the Milestone-2 of this application to support more wallets.

mmagician replied:
> Based on this SDK, we will use TypeScript to rewrite and improve this SDK in the Milestone-2 of this application to support more wallets.

Could you update the application document and include this? Please list the wallets that your SDK would support (at least one should be a Polkadot wallet)

AKACoder replied:
@mmagician Thanks for the review!

The deliverables have been updated.

We add the multi-chain & wallets support in milestone 2 and milestone 3's deliverable, and list all the chains and wallets we will be supported on this application.

mmagician replied:
Great!

AKACoder commented:
@mmagician Thanks for the review and feedback!

We replied the questions, please review.

AKACoder commented:
@Noc2 @mmagician   

Hello, we added supported wallet list include polkadot{.js} extension a week ago, and updated the application documents.  
Please review it when you have time to see if there are any other things that need to be updated or added.

Thanks!

mmagician reviewed (approved):
Apologies for late reply. I appreciate the updates regarding multi chain support and I'm looking forward to seeing your deliverables!

alxs commented:
Hey @AKACoder. Care to share a quick status update on the project, together with an ETA if possible? The first milestone is now delayed by several months. If you're no longer interested, we can terminate the agreement.

AKACoder commented:
> Hey @AKACoder. Care to share a quick status update on the project, together with an ETA if possible? The first milestone is now delayed by several months. If you're no longer interested, we can terminate the agreement.

@alxs We are sorry that we have been delayed for such a long time, and sorry for late reply. 
We encountered some difficulties in terms of project personnel and when using ink technology, which took us a lot of time and energy, but we have now solved these problems already. 

We are solving some of the final testing issues and final preparations for related documents, and will deliver milestone-1 within 3 days. The deliverables involved in milestone 1 will be mainly stored in this repository: [smart-contracts](https://github.com/SealSC/smart-contracts)

alxs commented:
No worries and thanks for the quick delivery. We'll try and look into it soon, but we're seeing a large number of submissions in the last few days (from belated teams we reach out to, such as yours), so it might take a bit longer. Please bear with us and don't hesitate to start working on the next milestone already.

AKACoder commented:
> No worries and thanks for the quick delivery. We'll try and look into it soon, but we're seeing a large number of submissions in the last few days (from belated teams we reach out to, such as yours), so it might take a bit longer. Please bear with us and don't hesitate to start working on the next milestone already.

Thanks for your reply! We will keeping working on the next milestones.

alxs commented:
Hey @AKACoder, any updates on the next milestone?

AKACoder commented:
> Hey @AKACoder, any updates on the next milestone?

@alxs Thanks for your attention！

The coding and testing of our second milestone has been completed in this repository: [multi-chain-js](https://github.com/SealSC/multi-chain-js) . 
We are working on finishing the document for now.
It is expected that the second milestone will be delivered out next week.

takahser commented:
Hi @AKACoder 
Has there been any activity on the implementation of M3 recently?
Is there any way we can support you?

alxs commented:
@AKACoder any updates? It looks like there hasn't been any activity on the [project repository](https://github.com/SealSC/multi-chain-js/branches) for the past 3 months. Note that if we don't hear from you within the next two weeks, we will assume you're no longer interested and terminate the grant.

AKACoder commented:
> @AKACoder any updates? It looks like there hasn't been any activity on the [project repository](https://github.com/SealSC/multi-chain-js/branches) for the past 3 months. Note that if we don't hear from you within the next two weeks, we will assume you're no longer interested and terminate the grant.

@alxs Sorry to reply you so late.
The M3 development is in a private repository, which is not yet open source. We plan to open source the code, launch the product, and deliver the M3 by the end of February.

AKACoder commented:
> Hi @AKACoder Has there been any activity on the implementation of M3 recently? Is there any way we can support you?

@takahser Thanks for the concern. In fact we did have some problems in the development and debugging of the !ink contract, but we have solved these problems.   
We plan to complete the deliver M3 by the end of February.

Noc2 commented:
Sounds great. Thanks for the update!

AKACoder commented:
@takahser 
Hi, sorry to bother, we need some help for wasm supported testnet.

The situation is like this, our !ink contract is basically completed in the development phase on the local environment, but we cannot find a suitable testnet that supports wasm for on-chain testing.

We tried Shibuya, Beresheet, Jupiter and many other testnets. They are not support wasm well or don't have a valid faucet to claim test token or explorer missing. So this makes it difficult for us to finish our M3 as we are planing to build an online demo for users on testnet in the M3.

Do you have a suggested testnet that supports WASM well? 
Or, can we deploy smart contracts on the polkadot ecosystem via Moonbean's EVM?  

Thanks!

takahser commented:
Hi @AKACoder 
You can use [Canvas](https://gateway.pinata.cloud/ipns/dotapps.io/?rpc=wss%3A%2F%2Frococo-canvas-rpc.polkadot.io#/explorer), they have ink! smart contract support.
I hope that works out for you, lmk if you have more questions. 👍

AKACoder commented:
> Hi @AKACoder You can use [Canvas](https://gateway.pinata.cloud/ipns/dotapps.io/?rpc=wss%3A%2F%2Frococo-canvas-rpc.polkadot.io#/explorer), they have ink! smart contract support. I hope that works out for you, lmk if you have more questions. 👍

Awesome, it works well! Thanks a lot! @takahser

AKACoder commented:
> Hi @AKACoder You can use [Canvas](https://gateway.pinata.cloud/ipns/dotapps.io/?rpc=wss%3A%2F%2Frococo-canvas-rpc.polkadot.io#/explorer), they have ink! smart contract support. I hope that works out for you, lmk if you have more questions. 👍

@takahser Hello,  sorry to bother you again. The  [Canvas](https://gateway.pinata.cloud/ipns/dotapps.io/?rpc=wss%3A%2F%2Frococo-canvas-rpc.polkadot.io#/explorer) has not been working for several hours, will it recover soon? Or do you have any other wasm supported chain recommendations?

alxs commented:
@AKACoder it's just the IPFS hosted version of polkadot.js which seems to be unreachable. You can use https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frococo-canvas-rpc.polkadot.io#/explorer

AKACoder commented:
> @AKACoder it's just the IPFS hosted version of polkadot.js which seems to be unreachable. You can use https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frococo-canvas-rpc.polkadot.io#/explorer

Thanks!

semuelle commented:
Hey @AKACoder, it's been a while since your planned delivery end of February. Would you mind providing an update?

alxs commented:
@AKACoder note that if we don't hear from you within the next 2 weeks, we'll assume you're no longer interested and terminate the grant.
